format-delimited/src/main/java/io/cdap/plugin/format/delimited/common/DataTypeDetectorUtils.java

```diff
@@ -38,10 +40,20 @@
    * @param dataTypeDetectorStatusKeeper Object that keeps the state of automated data type detection process.
    * @param columnNames Column names.
    * @param rowValues Row values.
+   * @param collector Failure Collector to store failures
    */
   public static void detectDataTypeOfRowValues(Map<String, Schema> override,
                                                DataTypeDetectorStatusKeeper dataTypeDetectorStatusKeeper,
-                                               String[] columnNames, String[] rowValues) {
+                                               String[] columnNames, String[] rowValues,
+                                               @Nullable FailureCollector collector) {
```

Comment:
why do we make it nullable?

Reply:
The reason for this is that the calling function [getDefaultSchema](https://github.com/cdapio/hydrator-plugins/blob/dba265837f986b7bb95b155b07e73266a51d3560/format-delimited/src/main/java/io/cdap/plugin/format/delimited/input/DelimitedInputFormatProvider.java#L136-L186) receives FailureContext as a nullable param. If context is null, context.getFailureCollector cannot be called, so collector will be null. 
I have referred [this function](https://github.com/cdapio/hydrator-plugins/blob/develop/format-avro/src/main/java/io/cdap/plugin/format/avro/input/AvroInputFormatProvider.java#L116-L148). I'm not sure why getDefaultSchema has context as a Nullable param, while the only function that I could find calling it, getSchema in the same file, has context as a non-nullable parameter. The implementation of getDefaultSchema doesn't check context for null value before accessing it. So it seems that context should never have a null value in format-avro.
I have removed `@Nullable` from context param in the calling function getDefaultSchema and from collector in the above lines.

Reply:
Change discarded.